src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,165 @@
+import { Todo } from '../types/Todo';
+
+interface Props {
+  setError: React.Dispatch<React.SetStateAction<string>>,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  loader: number[],
+  setLoader:React.Dispatch<React.SetStateAction<number[]>>
+  todo: Todo,
+}
+
+export const TodoInfo:React.FC<Props> = ({
+  setError,
+  setTodos,
+  loader,
+  setLoader,
+  todo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [searchQuery, setSearchQuery] = useState(todo.title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [isEditing]);
+
+  const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+  };
+
+  const clickHandler = (id: number) => {
+    setLoader(prevState => [...prevState, id]);
+    deleteTodos(id)
+      .then(() => {
+        setTodos(prevState => (
+          prevState.filter(prevTodo => prevTodo.id !== id)
+        ));
+      })
+      .catch(() => showError('Unable to delete a todo', setError));
+  };
+
+  const statusClickHandler = (todoId : number, todoStatus: boolean) => {
+    setLoader(prevState => [...prevState, todoId]);
+    updateTodo(todoId, { completed: !todoStatus })
+      .then((updatedTodo) => {
+        setTodos(prevState => (
+          prevState.map(prevTodo => {
+            return (prevTodo.id === todoId) ? updatedTodo : prevTodo;
+          })
+        ));
+
+        setLoader(prevState => prevState.filter(id => id !== todoId));
+      })
+      .catch(() => showError('Unable to delete a todo', setError));
+  };
+
+  const blurHandler = () => {
+    setIsEditing(false);
+
+    switch (searchQuery) {
+      case '':
+        setLoader(prevState => [...prevState, todo.id]);
+        deleteTodos(todo.id)
+          .then(() => {
+            setTodos(prevState => (
+              prevState.filter(prevTodo => prevTodo.id !== todo.id)
+            ));
+          })
+          .catch(() => showError('Unable to delete a todo', setError));
+        break;
+
+      case todo.title:
+        return;
+
+      default:
+        setLoader(prevState => [...prevState, todo.id]);
+        updateTodo(todo.id, { title: searchQuery })
+          .then((updatedTodo) => {
+            setTodos(prevState => (
+              prevState.map(prevTodo => {
+                return (prevTodo.id === todo.id) ? updatedTodo : prevTodo;
+              })
+            ));
+
+            setLoader(prevState => prevState.filter(id => id !== todo.id));
+          })
+          .catch(() => showError('Unable to update a todo', setError));
+    }
+  };
+
+  const submitHandler = (event:React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    blurHandler();
+  };
+
+  const keyUpHandler = (
+    event :React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      setIsEditing(true);
+      setSearchQuery(todo.title);
+    }
```

Comment:
```suggestion
    if (event.key !== 'Escape') {
      return;
    }
    
    setIsEditing(true);
    setSearchQuery(todo.title);
```

Reply:
If be honnest i don`t understand why we must use return here.